plugins/in_kafka/in_kafka.c

```diff
@@ -234,6 +269,15 @@ static int in_kafka_init(struct flb_input_instance *ins,
         goto init_error;
     }
 
+    ctx->log_encoder = flb_log_event_encoder_create(FLB_LOG_EVENT_FORMAT_DEFAULT);
+
+    if (ctx->log_encoder == NULL) {
+        flb_plg_error(ins, "could not initialize log encoder");
+        flb_log_event_encoder_destroy(ctx->log_encoder);
```

Comment:
If `log_encoder` is `NULL` you shouldn't try to destroy it, what you need to do here is `goto init_error;`

Reply:
done